SPEC.md

```diff
@@ -0,0 +1,675 @@
+| **bondedAmount**                   | The amount of bonded stake.                                           |
+| **unbondedAmount**                 | The amount of unbonded stake.                                         |
+| **delegateAddress**                | The Ethereum address of the delegate.                                 |
+| **delegatedAmount**                | The amount of delegated stake.                                        |
+| **startRound**                     | The round the delegator becomes bonded and delegated to its delegate. |
+| **withdrawRound**                  | The round the delegator can withdraw its stake.                       |
+| **lastClaimTokenPoolsSharesRound** | The last round that the delegator claimed reward and fee pool shares. |
+
+#### Transcoder
+
+Transcoder information is stored in a transcoder structure.
+
+| Field                      | Description                                                                         |
+|----------------------------|-------------------------------------------------------------------------------------|
+| **lastRewardRound**        | The last round that the transcoder called received rewards as an active transcoder. |
+| **rewardCut**              | The percentage of inflationary rewards that a transcoder keeps.                     |
+| **feeShare**               | The percentage of transcode job fees that a transcoder shares with its delegators.  |
+| **pricePerSegment**        | The price per segment charged for a stream.                                         |
+| **pendingRewardCut**       | The pending reward cut for the next round a transcoder becomes active.              |
+| **pendingFeeShare**        | The pending fee share for the next round a transcoder becomes active.               |
+| **pendingPricePerSegment** | The pending price per segment for the next round a transcoder becomes active.       |
+| **tokenPoolsPerRound**     | The reward and fee pools for each round.                                            |
+
+## Faults
+
+Cases of transcoders exhibiting malicious behavior are faults. These transcoder faults can be broadly split into two categories: attributable and non-attributable faults.
+In cases of attributable faults, a watcher can submit a slash proof on-chain as evidence of a transcoder's malicious behavior. In cases of non-attributable faults, a transcoder's behavior can not be definitively considered
+malicious due to the possible failure of external dependencies.
+
+Attributable faults include:
+
+- **Missed verification**: The transcoder did not submit transcode proofs for challenged segments.
+- **Double claiming segments**: The transcoder claimed a segment more than once.
+
+Non-attributable faults include:
+
+- **Failed verification**: The transcoder failed verification for a challenged segment. This type of fault is non-attributable because the protocol cannot differentiate from the scenario in which
```

Comment:
Is a failed verification non-attributable? I could see if verification doesn't return, or returns a known error value that potentially being non-attributable, but if the oracle sends back a different result?

Reply:
Right - I was thinking about the scenario in which a transcoder fails verification due to non-deterministic transcoding results, but if part of the protocol participation expectations is that transcoders run their nodes using specifically defined architecture, then the failed verification fault can be considered attributable in that context.